app/pages/_app.js

```diff
@@ -32,24 +31,24 @@ export default class App extends NextApp {
   };
 
   render() {
-    const {Component, variables = {}, relayData} = this.props;
+    const {Component, variables = {}, router = {}, relayData} = this.props;
     const environment = createEnvironment(
       relayData,
       JSON.stringify({
         queryID: Component.query ? Component.query().params.name : undefined,
         variables
       })
     );
-
     return (
       <ErrorBoundary>
         <QueryRenderer
           environment={environment}
           query={Component.query}
-          variables={variables}
+          variables={{...router.query, ...variables}}
           render={({error, props}) => {
             if (error) return <div>{error.message}</div>;
-            if (props) return <Component {...props} />;
+            if (props)
+              return <Component {...props} router={this.props.router} />;
```

Comment:
```suggestion
              return <Component {...props} />;
```

Reply:
This is needed. I think the props being spread into the component are different props than `this.props`